File: PO/L3/pikpo3_java/dataprocessorlab/src/main/java/com/usatu/processor/DataProcessorService.java
package com.usatu.processor;

/**
 * В данном модуле реализуется класс с основной бизнес-логикой приложения.
 * Обычно такие модули / классы имеют в названии слово "Service".
 */

public class DataProcessorService {

    private final DataProcessorFactory procFactory;

    private final String datasource;

    public DataProcessorService(String datasource) {
        this.datasource = datasource;
        procFactory = new DataProcessorFactory();
    }

    /**
     * ВАЖНО! Обратите внимание, что метод runService использует только методы базового абстрактного класса DataProcessor
     *        и, таким образом, будет выполняться для любого типа обработчика данных (CSV или TXT), что позволяет в дальнейшем
     *        расширять приложение, просто добавляя другие классы обработчиков, которые, например, работают с базой данных или
     *        сетевым хранилищем файлов (например, FTP-сервером).
     */
    public void runService() {
        DataProcessor processor = procFactory.getProcessor(datasource);
        processor.run();
        processor.printResult();
    }

    public String getDatasource() {
        return datasource;
    }
}

File: PIKPO/L3/L3_Python/Processor/dataprocessor_service.py
from .dataprocessor_factory import DataProcessorFactory

"""
    В данном модуле реализуется класс с основной бизнес-логикой приложения. 
    Обычно такие модули / классы имеют в названии слово "Service".
"""


class DataProcessorService:

    def __init__(self, datasource: str):
        self.datasource = datasource
        # Инициализируем в конструкторе фабрику DataProcessor
        self.processor_fabric = DataProcessorFactory()

    def run_service(self) -> None:
        """ Метод, который запускает сервис обработки данных  """
        processor = self.processor_fabric.get_processor(self.datasource)        # Инициализируем обработчик
        if processor is not None:
            processor.run()
            processor.print_result()
        else:
            print('Nothing to run')
